modules/util/portal-tools-service-builder/src/main/resources/com/liferay/portal/tools/service/builder/dependencies/service_base_impl.ftl

```diff
@@ -243,6 +243,100 @@ import org.osgi.service.component.annotations.Reference;
 			return ${entity.variableName}Persistence.update(${entity.variableName});
 		}
 
+		<#if entity.hasExternalReferenceCode() && entity.hasLazyReference() && entity.hasEntityColumn("companyId") && stringUtil.equals(entity.name, "User") && entity.hasEntityColumn("batchImportStatus")>
+			/**
+			 * Adds the ${entity.humanName} to the database. Also notifies the appropriate model listeners.
+			 *
+			 * <p>
+			 * <strong>Important:</strong> Inspect ${entity.name}LocalServiceImpl for overloaded versions of the method. If provided, use these entry points to the API, as the implementation logic may require the additional parameters defined there.
+			 * </p>
+			 *
+			 * @param externalReferenceCode the external reference code of the ${entity.humanName}
+			 * @param companyId the primary key of the company
+			 * @return the ${entity.humanName} that was added
+			<#list serviceBaseExceptions as exception>
+			 * @throws ${exception}
+			</#list>
+			 */
+			@Override
+			public ${entity.name} add${entity.name}(String externalReferenceCode, long companyId) <#if (serviceBaseExceptions?size gt 0)>throws ${stringUtil.merge(serviceBaseExceptions)} </#if>{
+				${entity.name} ${entity.variableName} =
+					${entity.variableName}Persistence.create(
+						counterLocalService.increment());
+
+				${entity.variableName}.setExternalReferenceCode(externalReferenceCode);
+				${entity.variableName}.setCompanyId(companyId);
+
+				${entity.variableName}.setBatchImportStatus(1);
+
+				return ${entity.variableName}Persistence.update(${entity.variableName});
+			}
+		</#if>
+
+		<#if entity.hasExternalReferenceCode() && entity.hasLazyReference() && entity.hasEntityColumn("companyId") && (!entity.hasEntityColumn("groupId") || stringUtil.equals(entity.name, "Group")) && entity.hasEntityColumn("userId") && !stringUtil.equals(entity.name, "User") && entity.hasEntityColumn("batchImportStatus")>
+			/**
+			 * Adds the ${entity.humanName} to the database. Also notifies the appropriate model listeners.
```

Comment:
same comment about the comment, I think we should specify that the status will be 1